### PR TITLE
User & user info DB, backend apis

### PR DESCRIPTION
**Pull request by Toblerone0629**
- Create Database tables
    - User
    - personal_info
    - Role
    - Address
    - Contact
- Create user && auth routers
- Modify duplicate or similar logic to **services** folder
- Implement user & user info, auth APIs
    - user
        - getAllUsers (for all employees page)
        - getUserById (for user detail page)
        - delete (only set deleted to true, will still keep the data)
    - user info (personal_info)
        - create
        - update
        - both could create/update the whole table at once
    - auth
        - login
        - signup

### CODE REVIEW

File: server/db/models/personalInfo.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        trim: true,
+        lowercase: true,
+        required: "Email address is required",
+        validate: {
+            validator: function (value) {
+                return value.includes("@");
+            },
+        },
+    },
+    ssn: {
```
Comment (DDDDDeng2022) [inline server/db/models/personalInfo.js:37]: 还需要添加ssn的start date和end date

File: server/db/models/personalInfo.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    email: {
+        type: String,
+        unique: true,
+        trim: true,
+        lowercase: true,
+        required: "Email address is required",
+        validate: {
+            validator: function (value) {
+                return value.includes("@");
```
Comment (DDDDDeng2022) [inline server/db/models/personalInfo.js:33]: 后端其实不用再进行一遍邮箱的验证，前端form会先验证，合法的情况下才能成功提交

**Review by DDDDDeng2022 — CHANGES_REQUESTED** [review]
先加ssn的开始时间和结束时间再重新更新一下pr吧

File: server/db/models/personalInfo.js (hunk 1 above)
Comment (Toblerone0629) [reply server/db/models/personalInfo.js:37]: 看了一遍文档，目前没有看到有需要start date和end date的地方

File: server/db/models/personalInfo.js (hunk 2 above)
Comment (Toblerone0629) [reply server/db/models/personalInfo.js:33]: 前端和后端都有validation的情况才可以cover大部分的corner error case

**Review by DDDDDeng2022 — APPROVED** [review]

**Pull request merged**